tests/src/test/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClientTest.java

```diff
@@ -1,80 +1,124 @@
 import io.snyk.eclipse.plugin.properties.preferences.Preferences;
-import io.snyk.eclipse.plugin.properties.preferences.PreferencesUtils;
+import io.snyk.languageserver.LsBaseTest;
 import io.snyk.languageserver.protocolextension.messageObjects.HasAuthenticatedParam;
 import io.snyk.languageserver.protocolextension.messageObjects.SnykTrustedFoldersParams;
 
-class SnykExtendedLanguageClientTest {
-  private InMemoryPreferenceStore store = new InMemoryPreferenceStore();
-  private SnykExtendedLanguageClient cut;
-  private Preferences pref;
-
-  @BeforeEach
-  void setUp() {
-    store = new InMemoryPreferenceStore();
-    pref = Preferences.getInstance(store);
-    PreferencesUtils.setPreferences(pref);
-	cut = new SnykExtendedLanguageClient();
-  }
-
-  @Test
-  void testAddTrustedPathsAddsPathToPreferenceStore() {
-    SnykTrustedFoldersParams param = new SnykTrustedFoldersParams();
-    param.setTrustedFolders(new String[] { "trusted/path " });
-
-    cut.addTrustedPaths(param);
-
-    assertEquals("trusted/path", store.getString(Preferences.TRUSTED_FOLDERS, ""));
-  }
-
-  @Test
-  void testAddTrustedPathsDeduplicatesAndTrims() {
-    SnykTrustedFoldersParams param = new SnykTrustedFoldersParams();
-    param.setTrustedFolders(new String[] { "trusted/path", "trusted/path", " trusted/path " });
-
-    cut.addTrustedPaths(param);
-
-    assertEquals("trusted/path", store.getString(Preferences.TRUSTED_FOLDERS, ""));
-  }
-  
-  @Test
-  void testHasAuthenticatedSavesTokenAndApiURL() {
-	  HasAuthenticatedParam param = new HasAuthenticatedParam();
-	  param.setApiUrl("https://abc.d/ef");
-	  param.setToken("testToken");
-	  
-	  cut.hasAuthenticated(param);
-	  
-	  assertEquals(param.getToken(), pref.getAuthToken());
-	  assertEquals(param.getApiUrl(), pref.getEndpoint());
-  }
-  
-  @Test
-  void testHasAuthenticatedCanHandleEmptyApiURL() {
-	  HasAuthenticatedParam param = new HasAuthenticatedParam();
-	  param.setToken("testToken");
-	  
-	  cut.hasAuthenticated(param);
-	  
-	  assertEquals(param.getToken(), pref.getAuthToken());
-	  assertEquals(Preferences.DEFAULT_ENDPOINT, pref.getEndpoint());
-  }
-  
-  @Test
-  void testHasAuthenticatedUpdatesPrefToEmptyToken() {
-	  HasAuthenticatedParam param = new HasAuthenticatedParam();
-	  param.setToken("");
-	  
-	  cut.hasAuthenticated(param);
-	  
-	  assertEquals(param.getToken(), pref.getAuthToken());
-	  assertEquals(Preferences.DEFAULT_ENDPOINT, pref.getEndpoint());
-  }
+class SnykExtendedLanguageClientTest extends LsBaseTest {
+	private InMemoryPreferenceStore store = new InMemoryPreferenceStore();
+	private SnykExtendedLanguageClient cut;
+	private Preferences pref;
+
+	@BeforeEach
+	protected void setUp() {
+		store = new InMemoryPreferenceStore();
+		pref = Preferences.getInstance(store);
+
+		// we don't want the wizard to pop up, so we set a dummy token
+		pref.store(Preferences.AUTH_TOKEN_KEY, "dummy");
```

Comment:
in this case maybe we don't need isTest then ?

Reply:
we need it - this is not sufficient due to initialization order :(